railties/test/application/mailer_previews_test.rb

```diff
@@ -253,8 +253,70 @@ def foo
       assert_no_match '<li><a href="/rails/mailers/notifier/bar">bar</a></li>', last_response.body
     end
 
-    test "mailer previews are reloaded from a custom preview_path" do
-      add_to_config "config.action_mailer.preview_path = '#{app_path}/lib/mailer_previews'"
+    test "mailer preview_path option is deprecated" do
+      begin
+        prev = ActionMailer::Base.preview_paths
+        ActiveSupport::Deprecation.silence do
+          ActionMailer::Base.preview_path = "#{app_path}/lib/mailer/previews"
+        end
+        assert_deprecated do
+          assert "#{app_path}/lib/mailer/previews", ActionMailer::Base.preview_path
+        end
+      ensure
+        ActionMailer::Base.preview_paths = prev
+      end
+    end
+
+    test "mailer preview_path= option is deprecated" do
+      begin
+        prev = ActionMailer::Base.preview_paths
+        assert_deprecated do
+          ActionMailer::Base.preview_path = "#{app_path}/lib/mailer/previews"
+        end
+        assert ["#{app_path}/lib/mailer/previews"], ActionMailer::Base.preview_paths
+      ensure
+        ActionMailer::Base.preview_paths = prev
+      end
+    end
+
+    test "mailer previews are loaded from custom a preview_paths when it's multiple" do
+      add_to_config "config.action_mailer.preview_paths = ['#{app_path}/lib/notifier_previews', '#{app_path}/lib/confirm_previews']"
+
+      ["notifier", "confirm"].each do |keyword|
+        mailer keyword, <<-RUBY
+          class #{keyword.camelize} < ActionMailer::Base
+            default from: "from@example.com"
+
+            def foo
+              mail to: "to@example.org"
+            end
+          end
+        RUBY
+
+        text_template "#{keyword}/foo", <<-RUBY
+          Hello, World!
+        RUBY
+
+        app_file "lib/#{keyword}_previews/notifier_preview.rb", <<-RUBY
+          class #{keyword.camelize}Preview < ActionMailer::Preview
+            def foo
+              #{keyword.camelize}.foo
+            end
+          end
+        RUBY
+      end
+
+      app("development")
+
+      get "/rails/mailers"
+      assert_match '<h3><a href="/rails/mailers/notifier">Notifier</a></h3>', last_response.body
+      assert_match '<li><a href="/rails/mailers/notifier/foo">foo</a></li>', last_response.body
+      assert_match '<h3><a href="/rails/mailers/confirm">Confirm</a></h3>', last_response.body
+      assert_match '<li><a href="/rails/mailers/confirm/foo">foo</a></li>', last_response.body
+    end
```

Comment:
This test can be combined with the existing one for loading previews from custom preview paths above.